File: src/main/java/io/slinkydeveloper/vertx/introduction/demo/core/HttpServerVerticle.java
package io.slinkydeveloper.vertx.introduction.demo.core;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Future;

public class HttpServerVerticle extends AbstractVerticle {

  private final int id;

  public HttpServerVerticle(int id) {
    this.id = id;
  }

  @Override
  public void start(Future<Void> startFuture) throws Exception {
    vertx.createHttpServer().requestHandler(req ->
      vertx
        .eventBus()
        .<String>send("randomquote.myapplication", "quote", ar -> {
          System.out.println(String.format("Handling from HttpServerVerticle thread: %d, id: %d", Thread.currentThread().getId(), id));
          if (ar.succeeded()) {
            req.response()
              .putHeader("content-type", "text/plain")
              .end(ar.result().body());
          } else {
            req.response()
              .setStatusCode(500)
              .putHeader("content-type", "text/plain")
              .end(ar.cause().getMessage());
          }
        })
    ).listen(8080, ar -> {
      if (ar.succeeded()) startFuture.complete();
      else startFuture.fail(ar.cause());
    });
  }
}

File: src/main/java/io/slinkydeveloper/vertx/introduction/demo/core/HttpServerVerticle.kt
package io.slinkydeveloper.vertx.introduction.demo.core

import io.vertx.kotlin.core.eventbus.sendAwait
import io.vertx.kotlin.core.http.listenAwait
import io.vertx.kotlin.coroutines.CoroutineVerticle
import kotlinx.coroutines.launch

class HttpServerVerticle(val id: Integer) : CoroutineVerticle() {

  override suspend fun start() {
    vertx.createHttpServer().requestHandler { req ->
      launch {
        val quoteMessage = vertx
          .eventBus()
          .sendAwait<String>("randomquote.myapplication", "quote")
        println("Handling from HttpServerVerticle thread: ${Thread.currentThread().id}, id: $id")
        req.response()
          .putHeader("content-type", "text/plain")
          .end(quoteMessage.body())
      }
    }.listenAwait(8080)
    println("HttpServerVerticle deployed")
  }
}
